extra/cs_package_manager.sh

```diff
@@ -917,13 +953,80 @@ do_install() {
     printf "\nDone\n\n"
 }
 
+prompt_user_for_os() {
+
+    # Prompt the user to select an operating system
+    echo "Please select an operating system to search for:"
+    os_options=("centos" "rhel" "rocky" "ubuntu" "debian")
+    select opt in "${os_options[@]}"; do
+        case $opt in
+            "centos" |  "rhel" | "rocky" )
+                distro_info=$opt
+                echo "What major version of $distro_info:"
+                short_options=("7" "8" "9")
+                select short in "${short_options[@]}"; do
+                         case $short in
+                            "7" | "8" | "9")
```

Comment:
worst case the user gets 0 results if they self pick it.  ill add a comment in the code
Was looking for a quick way when a mac user runs "cs_package_manager.sh check" since theres no /etc/os-release to auto detect what OS / version to search the mariadb repos. not perfect as this will require update in the future when new OS/versions are released.
Ideally theres an easy endpoint or list to get from our downloads site of the OS, version combinations that we support as opposed to parsing the download links